micro-rdk-installer/README.md

```diff
@@ -1,12 +1,63 @@
 
-When completed, this will be a CLI that allows a user to flash a build of Micro-RDK, along with
+A CLI that allows a user to flash a build of Micro-RDK, along with
 their robot's credentials and their wifi information, directly to their esp32 without requiring
 installation of ESP-IDF, Rust, or Python.
 
-For now, this only has the option of generating a binary of data representing the esp32's Non-Volatile 
-Storage (NVS) Partition and containing wifi and robot credentials. One could then potentially flash just
-the NVS partition to their esp32 using tools provided by ESP-IDF. 
+## Build Instructions
+
+1. `cargo build`
+2. ELF will be in target/debug
+
+## Usage
+
+```text
+A CLI that can flash a compilation of micro-RDK directly to an ESP32 provided configuration information
+
+Usage: micro-rdk-installer [COMMAND]
+
+Commands:
+  write-flash           Flash a pre-compiled binary with the micro-RDK server directly to an ESP32 connected to
+                            your computer via data cable
+  write-credentials     Write Wi-Fi and robot credentials to the NVS storage portion of a pre-compiled binary
+                            running a micro-RDK server
+  create-nvs-partition  Generate a binary of a complete NVS data partition that conatins Wi-Fi and security
+                            credentials for a robot
+  help                  Print this message or the help of the given subcommand(s)
+
+Options:
+  -h, --help  Print help
+```
+
+### Quick use
+
+1. Find your robot part at https://app.viam.com. Then navigate to the Setup tab
+2. Follow Step 1 under the instructions for Linux (regardless of your operating system) to download the Viam app config for your robot
+3. Run: `./micro-rdk-installer write-flash --app-config=<file path to the viam.json file downloaded in previous step>`
+    1. To see the micro-RDK server logs through the serial connection, add `--monitor`
+    2. If the program cannot auto-detect the serial port to which your ESP32 is connected, you may be prompted to select the correct one among a list
+
+## Common Problems
+
+### Linux Port Permissions
+
+If a "Permission Denied" or similar port error occurs, first check the connection of the ESP32 to the machine's USB port. If 
+connected and the error persists, run `sudo usermod -a -G dialout $USER` to add the current user to the `dialout` group, then try again.
+
+### MacOS Executable Permissions
+
+When using a machine running a version of MacOS, the user will be blocked from running the executable. To fix this, **Control+Click** the binary in Finder and then, in the following two prompts select **Open**. Close whatever terminal window this opens
+and then you should be able to run the installer.
+
+### Error: FlashConnect
+
+This may occur because the serial port chosen if/when prompted is incorrect. However, if the correct port has been selected, try the following:
+
+1. Run the installer as explained above
+2. When prompted to select a serial port
```

Comment:
What does prompt message look like? In docs we would put in bold: like **select a serial port** if the message said "select a serial port" in terminal ui

Reply:
I believe there isn't one. The prompt is created by an underlying library function and I found it difficult to hook into it. I'll give it another shot and then update the instructions